### PR TITLE
Add SecretDetails to SessionHostStartInfo

### PR DESCRIPTION
**Pull request by jotavalenciano**
To support the deployment of Secrets in the VMs we need this new class in the SessionHostStartInfo

### CODE REVIEW

File: AgentInterfaces/SecretDetail.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+        /// <summary>
+        /// The version of this secret in the internal Key Vault
+        /// </summary>
+        public string Version { get; set; }
+
+        /// <summary>
+        /// Expiration date of this secret
+        /// </summary>
+        public string ExpirationDate { get; set; }
```
Comment (emmayspark) [inline AgentInterfaces/SecretDetail.cs:28]: why we have 2 different data type for Expiration date?

File: AgentInterfaces/SessionHostsStartInfo.cs [hunk 2]
```diff
@@ -86,6 +86,11 @@ public class SessionHostsStartInfo
         /// </summary>
         public CertificateDetail[] GameCertificates { get; set; }
 
+        /// <summary>
+        /// List of secrets we want to deploy on each container
```
Comment (emmayspark) [inline AgentInterfaces/SessionHostsStartInfo.cs:90]: why on each container? are we planning to support only on container?

File: AgentInterfaces/SecretDetail.cs [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+        /// <summary>
+        /// Expiration date of the secret.
```
Comment (emmayspark) [inline AgentInterfaces/SecretDetail.cs:31]: maybe we should add this is for UTC and why UTC type is needed?

File: AgentInterfaces/SecretDetail.cs (hunk 1 above)
Comment (jotavalenciano) [reply AgentInterfaces/SecretDetail.cs:28]: I copied the same from CertificateDetail class. I want to make it consistent

File: AgentInterfaces/SessionHostsStartInfo.cs (hunk 2 above)
Comment (jotavalenciano) [reply AgentInterfaces/SessionHostsStartInfo.cs:90]: I copied the same from CertificateDetail class. I want to make it consistent. But your are right. Maybe container is not the right word

File: AgentInterfaces/SecretDetail.cs (hunk 3 above)
Comment (jotavalenciano) [reply AgentInterfaces/SecretDetail.cs:31]: I copied the same from CertificateDetail, but you are right. Having both of them is not great

File: AgentInterfaces/SecretDetail.cs [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+// Licensed under the MIT License.
+
+namespace Microsoft.Azure.Gaming.AgentInterfaces
+{
+    using System;
+
+    public class SecretDetail
+    {
+        /// <summary>
+        /// Identifier secret
```
Comment (dgkanatsios) [inline AgentInterfaces/SecretDetail.cs:11]: nit: Secret identifier

File: AgentInterfaces/SecretDetail.cs [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+        /// <summary>
+        /// The version of this secret in the internal Key Vault
+        /// </summary>
+        public string Version { get; set; }
+
+        /// <summary>
+        /// Expiration date of this secret
+        /// </summary>
+        public string ExpirationDate { get; set; }
+}
```
Comment (dgkanatsios) [inline AgentInterfaces/SecretDetail.cs:29]: there's a } missing, as depicted here [https://github.com/PlayFab/MpsAgent/actions/runs/6399628016/job/17371970994?pr=171](https://github.com/PlayFab/MpsAgent/actions/runs/6399628016/job/17371970994?pr=171)

Comment (jotavalenciano) [reply AgentInterfaces/SecretDetail.cs:30]: OMG yes. Don't know why I missed that

File: AgentInterfaces/SecretDetail.cs (hunk 3 above)
Comment (jotavalenciano) [reply AgentInterfaces/SecretDetail.cs]: Removing the second parameter

**Review by dgkanatsios — APPROVED** [review]
:shipit:

**Pull request merged**